packages/stablestudio-plugin-comfyui/src/index.ts

```diff
@@ -0,0 +1,615 @@
+          ).then((r) => r.blob()),
+          input: input,
+        } as StableStudio.StableDiffusionImage);
+      }
+      console.log(images);
+      return images;
+    } else {
+      return undefined;
+    }
+  } else {
+    return undefined;
+  }
+};
+
+const uploadImage = async (apiUrl: string, image: Blob, filename: string) => {
+  const blob = new Blob([image], { type: "image/png" });
+  const formData = new FormData();
+  formData.append("image", blob, filename);
+  const response = await fetch(`${apiUrl}/upload/image`, {
+    method: "POST",
+    body: formData,
+  });
+  if (response.ok) {
+    const json = await response.json();
+    return {
+      name: json.name,
+      subfolder: json.subfolder,
+      type: json.type,
+    };
+  } else {
+    return {
+      error: "Error",
+    };
+  }
+};
+
+const randomFilename = () => {
+  return (
+    Math.random().toString(36).substring(2, 15) +
+    Math.random().toString(36).substring(2, 15)
+  );
+};
+
+const randomSeed = () => {
+  return Math.floor(Math.random() * 1000000000);
+};
+
+const getDefaultInput = () => ({
+  width: 512,
+  height: 512,
+
+  cfgScale: 7,
+  steps: 30,
+  sampler: { id: "0", name: "euler" },
+  model: "0", //hack to select first model from list
+  strength: 0.75,
+});
+
+export const createPlugin = StableStudio.createPlugin<{
+  settings: {
+    apiUrl: StableStudio.PluginSettingString;
+  };
+}>(({ get, set }) => ({
+  manifest: {
+    name: "ComfyUI",
+    author: "hlky",
+    link: "https://github.com/comfyanonymous/ComfyUI",
+    icon: `${window.location.origin}/DummyImage.png`,
+    version: "0.0.1",
+    license: "MIT",
+    description: "ComfyUI plugin for StableStudio",
+  },
+  getStableDiffusionSamplers: async () => {
+    return await getSamplers(get().settings.apiUrl.value as string).then(
+      (samplers) =>
+        samplers.map((s: string) => ({
+          id: `${samplers.indexOf(s)}`,
+          name: s,
+        }))
+    );
+  },
+
+  getStableDiffusionModels: async () => {
+    const models = getModels(get().settings.apiUrl.value as string).then(
+      (models) =>
+        models.map((model: string) => ({
+          id: `${models.indexOf(model)}`,
+          name: model,
+        }))
+    );
+    return models;
+  },
+
+  createStableDiffusionImages: async (options) => {
+    const count = options?.count ?? 1;
+    const apiUrl = get().settings.apiUrl.value as string;
+    const models = await getModels(apiUrl);
+    const model = models[options?.input?.model as string];
+    const defaultStableDiffusionInput = getDefaultInput();
+    const input = {
+      ...defaultStableDiffusionInput,
+      ...options?.input,
+    };
+    input.model = model;
+    if (!input.model) {
+      return undefined;
+    }
+
+    const prompt =
+      input.prompts
+        ?.filter((p) => p.weight && p.weight > 0)
+        .map((p) => `(${p.text}:${p.weight})`)
```

Comment:
Wraps each input.prompt in (text:weight), there's only one prompt box but it should support the `Prompt strength`